spec/jobs/embargo_expiry_job_spec.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+RSpec.describe EmbargoExpiryJob do
+  subject { described_class }
+
+  let(:past_date) { 2.days.ago }
+
+  let(:embargoed_work) { create(:embargoed_work) }
+
+  let!(:work_with_expired_embargo) do
+    build(:work, embargo_release_date: past_date.to_s).tap do |work|
+      work.save(validate: false)
+    end
+  end
+
+  let!(:file_set_with_expired_embargo) do
+    build(:file_set, embargo_release_date: past_date.to_s).tap do |file_set|
+      file_set.save(validate: false)
+    end
+  end
+
+  describe '#records_with_expired_embargos' do
+    it 'returns all records with expired embargos' do
+      records = described_class.new.records_with_expired_embargos
+      expect(records).to include work_with_expired_embargo, file_set_with_expired_embargo
+      expect(records).not_to include embargoed_work
+    end
+  end
+
+  describe '#perform' do
+    it "Enqueues an Expire Embargo job for each expired record" do
+      expect { described_class.perform_now }.to change(ActiveJob::Base.queue_adapter.enqueued_jobs, :size).by(described_class.new.records_with_expired_embargos.count)
```

Comment:
I think it may be nicer to use the `have_enqueued_job` matcher instead of `change`.
https://relishapp.com/rspec/rspec-rails/docs/matchers/have-enqueued-job-matcher

Reply:
Nevermind.  I don't see any examples of this matcher in the Hyrax codebase yet.  Maybe expecting calls to `perform_later` then: https://github.com/samvera/hyrax/blob/130c4e600318a9194725b35dd0fd6e19e5108dd9/spec/config/hyrax_events_spec.rb#L9